### ISSUE TITLE
flightaxis SITL - incorrect current reading when negative values

### ISSUE THREAD

**Issue by nhadrian** [ArduPilot/MissionPlanner#2535]
#### Issue details
When I'm using SITL with RF9 and flighaxis controller, negative current values in RF9 (when motor is braking) gets false high readings in Mission Planner. This causes false drop in battery percentage. It seems like an issue with signs and flightaxis/MP link.

RF9 current reading (-1.1A):
![unknown](https://user-images.githubusercontent.com/5570527/103976491-234ae100-5177-11eb-8fe2-0dc09ad36b8a.png)

Mission Planner current reading (327,7A): 
![unknown2](https://user-images.githubusercontent.com/5570527/103976496-25ad3b00-5177-11eb-82a7-48fa0407b1e2.png)

#### Version
1.3.74.1

#### Platform
[  ] All
[  ] AntennaTracker
[  ] Copter
[ X ] Plane
[  ] Rover
[  ] Sub

#### Airframe type
plane, quadplane

#### Hardware type
SITL / flightaxis

**Comment by meee1** [comment]
do you have a tlog of this as well?

**Comment by nhadrian** [comment]
Yes, in this short tlog after takeoff (at about 2/3 time of log length) you can see multiple times when current in RealFlight changes to negative, MP shows 327,7A.
[2021-01-08 09-11-30.zip](https://github.com/ArduPilot/MissionPlanner/files/5785993/2021-01-08.09-11-30.zip)

**Comment by meee1** [comment]
ok so this looks like an ardupilot bug
the field over mavlink is signed, so to get that value is bad
if you graph current and battery remaining, there is a large jump in capacity
![image](https://user-images.githubusercontent.com/511568/103993310-5d949c80-51d0-11eb-8fdc-c11f7e65804a.png)

i would suggest posting the issue in the arudpilot issue list, and link to this issue

**Comment by IamPete1** [comment]
I'm fairly sure that the Realflight NavGuides are just nonsense and that there is no underlying bug in AP. This can probably be closed.

https://github.com/ArduPilot/ardupilot/issues/16272#issuecomment-761693444
